metricflow/dataset/convert_data_source.py

```diff
@@ -331,7 +331,7 @@ def _create_identifier_instances(
                     expr = sub_id.expr
                     if expr is None:
                         assert sub_id.name is not None
-                        expr = sub_id.name.element_name
+                        expr = sub_id.name
                     sub_id_name = sub_id.ref or sub_id.name
                     assert sub_id_name, f"Sub-identifier {sub_id} must have 'name' or 'ref' defined"
```

Comment:
'ref' in the string assertion should be 'reference'

Reply:
So we should change it there too then.

Reply:
You're suggesting that we should change how customers define YAML files?

Reply:
Haha no, I know better than to suggest that to you. I was more meaning change it in the process from the YAML field to when it gets into MetricFlow. Although, I suppose we could change it directly in the YAML. There will be breaking changes to the YAMLs at some point so this could be one of them. Three options perhaps:

1) change directly in the YAML
2) translate it to `reference` within MetricFlow so it's consistent throughout.
3) Deal with the inconsistency and do nothing about it.